garden-backend-service/src/api/routes/users.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any
+
+from fastapi import APIRouter, Depends, status
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from src.api.dependencies.auth import authed_user
+from src.api.dependencies.database import get_db_session
+from src.api.schemas.user import UserMetadataResponse, UserUpdateRequest
+from src.models import User
+from src.models._associations import users_saved_gardens
+
+router = APIRouter(prefix="/users")
+
+
+async def _get_saved_garden_dois(user: User, db: AsyncSession) -> list[str]:
+    from src.models import Garden  # import here to avoid circular imports
+
+    result = await db.execute(
```

Comment:
I think it's exactly equivalent to what you have but FYI the `.scalars` method instead of `execute` could skip the row objects